modules/codegen/src/smithy4s/codegen/Renderer.scala

```diff
@@ -338,11 +330,11 @@ private[codegen] class Renderer(compilationUnit: CompilationUnit) { self =>
       recursive: Boolean,
       hints: List[Hint]
   ): RenderResult = {
-    val decl = s"case class $name(${renderArgs(fields)})"
-    val imports = fields.foldMap(_.tpe.imports) ++ syntaxImport
+    val decl = render"case class $name(${renderArgs(fields)})"
+    val imports = decl.imports ++ syntaxImport
     lines(
       if (hints.contains(Hint.Error)) {
-        block(s"$decl extends Throwable") {
+        block(s"${decl.line} extends Throwable") {
```

Comment:
as you correctly inferred, let's make `block` take a `Line`, and have a method overload that'd take a `String`